front/src/modules/powerRestriction/helpers/formatPowerRestrictionRangesWithHandled.ts

```diff
@@ -29,49 +32,42 @@ export const formatPowerRestrictionRanges = (
         return null;
       }
       return {
-        start: mmToM(stepsPathPositions[startStep]),
-        stop: mmToM(stepsPathPositions[stopStep]),
-        code: powerRestriction.value,
+        position: {
+          start: stepsPathPositions[startStep],
+          end: stepsPathPositions[stopStep],
+        },
+        value: { powerRestriction: powerRestriction.value },
       };
     })
   );
 
 /** Format power restrictions data to be used in simulation results charts */
-export const addHandledToPowerRestrictions = (
-  powerRestrictionRanges: Omit<SimulationPowerRestrictionRange, 'handled'>[],
+export const addHandledToPowerRestrictionsAndConvertPositions = (
```

Comment:
Do you think it's better ?

```suggestion
export const convertPowerRestrictionsAndCheckCompatibility = (
```

Reply:
done